src/ast.rs

```diff
@@ -73,7 +78,7 @@ impl Location {
           }
         }
       }
-      unreachable!()
+      return format!("line {}, col {}", 0, 0);
```

Comment:
nit: prefer returning using the 'last statement without trailing semicolon' rather than with `return`

```suggestion
      format!("line {}, col {}", 0, 0)
```